bson-record-codec/src/main/org/bson/codecs/record/RecordCodec.java

```diff
@@ -83,6 +85,10 @@ Object getValue(final Record record) throws InvocationTargetException, IllegalAc
         @SuppressWarnings("deprecation")
         private static Codec<?> computeCodec(final RecordComponent component, final CodecRegistry codecRegistry) {
             var codec = codecRegistry.get(toWrapper(component.getType()));
+            if (codec instanceof Parameterizable parameterizableCodec
```

Comment:
What is this magic? assigning variables in an if statement :exploding_head:

Reply:
Not just one but two!